src/create-draggable.ts

```diff
@@ -0,0 +1,113 @@
+import { Listeners, useDragDropContext } from "./drag-drop-context";
+import {
+  elementLayout,
+  noopTransform,
+  Transform,
+  transformsAreEqual,
+} from "./layout";
+import { transformStyle } from "./style";
+import {
+  createEffect,
+  createRenderEffect,
+  createSignal,
+  onCleanup,
+  onMount,
+  Setter,
+} from "solid-js";
+
+interface Draggable {
+  (element: HTMLElement): void;
+  ref: Setter<HTMLElement | null>;
+  get isActiveDraggable(): boolean;
+  get dragActivators(): Listeners;
+  get transform(): Transform;
+}
+
+const createDraggable = (
+  id: string,
+  data: Record<string, any> = {}
+): Draggable => {
+  const [state, { addDraggable, removeDraggable, draggableActivators }] =
+    useDragDropContext()!;
+  const [node, setNode] = createSignal<HTMLElement | null>(null);
+
+  onMount(() => {
+    const resolvedNode = node();
+
+    if (resolvedNode) {
+      addDraggable({
+        id,
+        node: resolvedNode,
+        layout: elementLayout(resolvedNode),
+        data,
+      });
+    }
+  });
+  onCleanup(() => removeDraggable(id));
+
+  const isActiveDraggable = () => state.active.draggable === id;
+  const transform = () => {
+    return state.draggables[id]?.transform || noopTransform();
+  };
+
+  const draggable = Object.defineProperties(
```

Comment:
Per [Solid docs](https://www.solidjs.com/docs/latest/api#special-jsx-attributes), users should define the directives in a separate declaration file. It didn't work too well for me, so I reverted back to just using the directive as a function with `ref`.

Potentially this could be declared in a declaration file attached to `solid-dnd` but if the user names his draggable differently, this won't work.

Reply:
Thanks for the info. Let's leave it as would be brittle to do anything in solid-dnd itself. If it becomes an issue for others we can always add a note to the docs with a link to Solid guidance.